logbook-spring-webflux/src/main/java/org/zalando/logbook/spring/webflux/BufferingServerHttpRequest.java

```diff
@@ -49,10 +50,11 @@ public Flux<DataBuffer> getBody() {
     private Publisher<? extends DataBuffer> bufferingWrap(Publisher<? extends DataBuffer> body) {
         if (serverRequest.shouldBuffer()) {
             return Flux
-                    .from(DataBufferCopyUtils.wrapAndBuffer(body, serverRequest::buffer))
-                    .doOnComplete(writeHook);
+                .from(DataBufferCopyUtils.wrapAndBuffer(body, serverRequest::buffer))
+                .doOnComplete(writeHook);
         } else {
-            return body;
+            return Mono.fromRunnable(writeHook)
```

Comment:
That's an interesting finding! Could you please check what's failing in terms of coverage in the pipeline?

Reply:
Branch coverage is now under 100% in `LogbookWebFilter` in line 41:
`if (s instanceof Logbook.ResponseWritingStage) ((Logbook.ResponseWritingStage) s).write();`

Before we had 100%, as a result of this issue. So in this line `s` was still an instance of `Logbook`, as the request never was processed. 
Now `s instanceof Logbook.ResponseWritingStage` result to `true` all the time.

I believe the `if` in line 41 will result to `true` all the time, if all requests are processed, which they should be. So not sure how to reach 100% coverage again.

Reply:
Not an ideal solution, but how about adding a mocked unit test for the LogbookWebFilter that'll check that `write()` is not called when the stage object is wrong? 

Something like
```

    final Logbook logbook = mock(Logbook.class);
    final ServerWebExchange exchange = mock(ServerWebExchange.class);
    final ServerWebExchange.Builder builder = mock(ServerWebExchange.Builder.class);
    final ServerHttpRequest request = mock(ServerHttpRequest.class);
    final ServerHttpResponse response = mock(ServerHttpResponse.class);
    final WebFilterChain chain = mock(WebFilterChain.class);
    final Logbook.RequestWritingStage stage = mock(Logbook.RequestWritingStage.class);
    final  LogbookWebFilter underTest = new LogbookWebFilter(logbook);

    @Test
    void shouldNotCallWriteOnInappropriateStage() throws Throwable {
        when(logbook.process(any())).thenReturn(stage);
        when(exchange.getRequest()).thenReturn(request);
        when(exchange.getResponse()).thenReturn(response);
        when(exchange.mutate()).thenReturn(builder);
        when(builder.request(any(ServerHttpRequest.class))).thenReturn(builder);
        when(builder.response(any())).thenReturn(builder);
        when(builder.build()).thenReturn(exchange);
        when(chain.filter(any())).thenReturn(Mono.empty());

        underTest.filter(exchange, chain).block();

        verify(stage, never()).write();
    }
```

Reply:
Thank you (and sorry I didn't approve the pipeline on Friday). It's failing now due to a vulnerability, that's already fixed. If you pull the latest main branch it should be green. I'll merge the PR after that

Reply:
I merged the latest changes from the main branch